src/jsTest/kotlin/it/krzeminski/githubactionstyping/AjvSchemaValidationTest.kt

```diff
@@ -0,0 +1,100 @@
+package it.krzeminski.githubactionstyping
+
+import ajv.Ajv
+import ajv.ValidateFunction
+import io.kotest.assertions.withClue
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.booleans.shouldBeTrue
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+import io.kotest.mpp.env
+import node.buffer.BufferEncoding.Companion.utf8
+import node.fs.exists
+import node.fs.existsSync
+import node.fs.opendir
+import node.fs.readFile
+import node.fs.readdirSync
+import node.fs.stat
+import node.path.path
+import ajv.Options as AjvOptions
+
+private val schemaFile = env("schemaFile")!!
+private val catalogDir = env("catalogDir")!!
+private val externalDir = env("externalDir")!!
+private val goodDir = env("goodDir")!!
+private val badDir = env("badDir")!!
+
+private lateinit var validate: ValidateFunction
+
+/**
+ * Validate good and bad typings against the Ajv schema validator.
+ */
+class AjvSchemaValidationTest : FunSpec({
+    beforeSpec {
+        withClue("catalogDir should be a non-empty directory") {
+            exists(catalogDir).shouldBeTrue()
+            stat(catalogDir).isDirectory().shouldBeTrue()
+            opendir(catalogDir).read().await().shouldNotBeNull()
+        }
+
+        validate = Ajv(AjvOptions(strict = true)).compile(
+            JSON.parse(readFile(schemaFile, utf8))
+        )
+    }
+
+    if (existsSync(catalogDir)) {
+        withData(
+            nameFn = { "catalog typing: $it" },
+            readdirSync(catalogDir).asSequence()
```

Comment:
FYI, one day we may switch to a platform-agnostic way of reading the resources (working on it in snakeyaml-engine-kmp, it already works there - [ref](https://github.com/krzema12/snakeyaml-engine-kmp/blob/main/buildSrc/src/main/kotlin/buildsrc/tasks/ConvertCommonTestResourcesToKotlin.kt)), but for now we can stay with how things are.